hydroflow/src/scheduled/handoff/vector.rs

```diff
@@ -50,12 +55,25 @@ where
         (*self.deque).borrow_mut().extend(&mut iter.0);
         iter
     }
+
+    fn swap(&self, _item: &mut Iter<I>) {
+        // TODO(justin): I'm not sure if there's a sane implementation for this.
+        todo!()
+    }
 }
 impl<T> CanReceive<VecDeque<T>> for VecHandoff<T> {
     fn give(&self, mut vec: VecDeque<T>) -> VecDeque<T> {
         (*self.deque).borrow_mut().extend(vec.drain(..));
         vec
     }
+
+    fn swap(&self, item: &mut VecDeque<T>) {
+        if (*self.deque).borrow().is_empty() {
+            std::mem::swap(&mut *(*self.deque).borrow_mut(), item);
+        } else {
+            (*self.deque).borrow_mut().extend(item.drain(..))
+        }
+    }
```

Comment:
Only on the push side?
I'm not convinced this is the best abstraction but it is good for now.

Reply:
Yeah good point, doesn't currently help much with perf so I'm going to just remove it for now.